File: src/main/java/logisticspipes/api/ILPPipeTile.java
package logisticspipes.api;

import net.minecraft.util.math.BlockPos;

/**
 * Public interface implemented by LP's Pipe tile
 */
public interface ILPPipeTile {

	/**
	 * Used to access the internal pipe logic This also can return the
	 * IRequestAPI or IRoutedPowerProvider
	 *
	 * @return the pipe
	 */
	ILPPipe getLPPipe();

	BlockPos getBlockPos();
}

File: common/logisticspipes/api/ILPPipeTile.kt
package logisticspipes.api

import net.minecraft.util.math.BlockPos

/**
 * Base interface for all pipe tiles in the LP mod
 */
interface ILPPipeTile {

	/**
	 * Used to access the internal pipe logic.
	 */
	fun getLPPipe(): ILPPipe

	/**
	 * Used to obtain the tile position
	 */
	fun getBlockPos(): BlockPos
}
